src/components/ClinicFinder.tsx

```diff
@@ -0,0 +1,490 @@
+    const RADIUS_OF_EARTH_IN_KM = 6371;
+
+    const dLat = distance(lat2, lat1);
+    const dLon = distance(lon2, lon1);
+
+    lat1 = toRadian(lat1);
+    lat2 = toRadian(lat2);
+
+    // Haversine Formula
+    const a =
+      Math.pow(Math.sin(dLat / 2), 2) +
+      Math.pow(Math.sin(dLon / 2), 2) * Math.cos(lat1) * Math.cos(lat2);
+    const c = 2 * Math.asin(Math.sqrt(a));
+
+    let finalDistance = RADIUS_OF_EARTH_IN_KM * c;
+
+    if (isMiles) {
+      finalDistance /= 1.60934;
+    }
+    return finalDistance;
+  };
+
+  const sortClinicsByDistance = (lat: number, long: number) => {
+    return clinicServices
+      .map((clinic, i) => {
+        return {
+          ...clinic,
+          mapsIdentifier: i.toString(),
+          distance: distanceBetweenCoordinates(
+            [lat, long],
+            [clinic.coordinatesLat, clinic.coordinatesLong],
+          ),
+        };
+      })
+      .sort((a, b) => {
+        return a.distance - b.distance;
+      })
+      .slice(0, 10);
+  };
+
+  const toggleInfoCard = async (id: number) => {
+    try {
+      await axios
+        .get(`http://127.0.0.1:3001/vet-practices/${id}`)
+        .then(response => {
+          setActiveInfoCardId(id);
+          setClinicServiceDetails(response.data);
+        });
+    } catch (e) {
+      console.log(e);
+    }
+  };
+
+  const resetClinicService = () => {
+    setActiveInfoCardId(null);
+  };
+
+  const renderDetails = clinic => {
+    if (clinicServiceDetails.type === ClinicType.emergencyRing) {
+      return (
+        <ClincDetailsEmergency
+          clinicServiceDetails={clinicServiceDetails}
+          backToList={() => {
+            resetClinicService();
+          }}
+        />
+      );
+    } else {
+      return (
+        <ClinicDetails
+          clinicServiceDetails={clinicServiceDetails}
+          backToList={() => {
+            resetClinicService();
+          }}
+        />
+      );
+    }
+  };
+
+  // Map actions
+  const backToMap = () => {
+    resetClinicService();
+    setShowItemList(false);
+  };
+
+  const clearPosition = () => {
+    setPositioning(false);
+    setUserGeoLocation(null);
+    setCustomPosition(null);
+  };
+
+  const moveToPosition = async () => {
+    await clearPosition();
+    await getGeoLoacation();
+    if (userGeoLocation) {
+      setCustomPosition({
+        lat: userGeoLocation.lat,
+        long: userGeoLocation.long,
+      });
+      setShowPin(false);
+      setPositioning(true);
+    }
+  };
+
+  const moveToSearchLocation = (placeId: string) => {
+    axios
+      .get(
+        `http://127.0.0.1:3001/mobile-app-frontend/vet-finder/location-search/geo-location?place-id=${placeId}`,
+      )
+      .then(response => {
+        setPositioning(true);
+        setCustomPosition({ lat: response.data.lat, long: response.data.long });
+        setShowPin(true);
+      })
+      .catch(e => {
+        console.log(e);
+      });
+  };
+
+  return (
```

Comment:
A lot of stuff is happening here, maybe divide into smaller components? Also this would be better for separation of styling and logic

Reply:
yes maybe, but lot of small stuff. idk if worth it. I'll check